dirty_equals/_other.py

```diff
@@ -145,3 +146,42 @@ def is_even(x):
 
     def equals(self, other: Any) -> bool:
         return self.func(other)
+
+
+class IsHash(DirtyEquals[str]):
+    """
+    A class that checks if a value is a valid common hash type.
```

Comment:
can you add some explanation of how this works - e.g. "Just uses a simple length and allowed characters regex".